arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+          "$ref": "#/definitions/ResourceProviderOperationDisplayInfo",
+          "description": "Gets or sets the display."
+        },
+        "origin": {
+          "description": "Gets or sets the origin.",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/ResourceProviderOperationProperties",
+          "description": "Gets or sets the properties."
+        }
+      }
+    },
+    "ResourceProviderOperationDisplayInfo": {
+      "description": "Contains properties related to the RP Operation Display Info.",
+      "type": "object",
+      "properties": {
+        "provider": {
+          "description": "Gets the provider.",
+          "type": "string",
+          "readOnly": true
+        },
+        "resource": {
+          "description": "Gets the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "operation": {
+          "description": "Gets or sets the operation.",
+          "type": "string"
+        },
+        "description": {
+          "description": "Gets or sets the description.",
+          "type": "string"
+        }
+      }
+    },
+    "ResourceProviderOperationProperties": {
+      "description": "Contains properties describing the Resource provider Operations.",
+      "type": "object",
+      "properties": {
+        "serviceSpecification": {
+          "$ref": "#/definitions/ServiceSpecification",
+          "description": "Gets or sets the service specification."
+        }
+      }
+    },
+    "ServiceSpecification": {
+      "description": "Contains properties related to a service specification.",
+      "type": "object",
+      "properties": {
+        "metricSpecifications": {
+          "description": "Gets or sets the metric specifications.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MetricSpecification"
+          }
+        },
+        "logSpecifications": {
+          "description": "Gets or sets the log specifications.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LogSpecification"
+          }
+        }
+      }
+    },
+    "MetricSpecification": {
+      "description": "Contains properties related to a metric specification.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Gets or sets the name.",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "Gets or sets the display name.",
+          "type": "string"
+        },
+        "displayDescription": {
+          "description": "Gets or sets the display description.",
+          "type": "string"
+        },
+        "unit": {
+          "description": "Gets or sets the unit.",
+          "enum": ["Bytes", "BytesPerSecond", "Count", "CountPerSecond", "Percent", "Seconds"],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "MetricUnit",
+            "modelAsString": "True"
```

Comment:
It will be breaking change in the generated code. You will have to update the  major version of the client library package in whichever language (C#.net, node.js, etc.)  you want to publish.
If you know that for some enums the set of allowed values are going to remain the same for a good time (~ few months) then it is ok to have modelAsString: false. (Statistically service teams have updated the major version of the package, once in 6 months either due to api-version change or some other reason).